src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        size = 0;
+        objects = new Object[10];
+    }
+
     @Override
     public void add(T value) {
+        if (size == objects.length) {
+            Object[] newObject = new Object[objects.length + (int) objects.length / 2];
+            for (int i = 0; i < objects.length; i++) {
+                newObject[i] = objects[i];
+            }
+            objects = newObject;
```

Comment:
The logic to resize the array should be extracted into a separate private method to avoid code duplication and adhere to the Single Responsibility Principle.